src/levanter/data/sharded_datasource.py

```diff
@@ -417,6 +427,24 @@ def open_shard_at_row(self, shard_name: str, row: int) -> Iterator[dict]:
             return iter(data[row:])
 
 
+class ParquetDataSource(ShardedDataSource[dict]):
```

Comment:
ideally the TextUrlDataSource would also work with parquet files. That is what we use for training configs typically

Reply:
Makes sense, I added a new test for this in #766